modules/msp/apm/log-service/query/log.service.histogram.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log_service
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ahmetb/go-linq/v3"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	monitorpb "github.com/erda-project/erda-proto-go/core/monitor/log/query/pb"
+	"github.com/erda-project/erda-proto-go/msp/apm/log-service/pb"
+	"github.com/erda-project/erda/modules/extensions/loghub/index/query"
+	"github.com/erda-project/erda/pkg/common/apis"
+)
+
+func (s *logService) HistogramAggregationFromMonitor(ctx context.Context, req *pb.HistogramAggregationRequest) (*pb.HistogramAggregationResponse, error) {
```

Comment:
How are FromMonitor and FromLoghub different?

Reply:
FromMonitor query from the union es storage, Fromloghub query from the old log-analysis es storage.